lib/iris/cube.py

```diff
@@ -3862,7 +3865,11 @@ def __eq__(self, other):
             if result:
                 # TODO: why do we use allclose() here, but strict equality in
                 #  _DimensionalMetadata (via util.array_equal())?
-                result = da.allclose(self.core_data(), other.core_data()).compute()
+                result = da.allclose(
+                    self.core_data(),
+                    other.core_data(),
+                    equal_nan=True,
+                ).compute()
```

Comment:
I know this wasn't you, but I think we would be better with only using Dask when a lazy array is involved, which will save overhead if everything is already NumPy. `np.allclose()` is intelligent enough to do this, and `as_concrete_data()` knows to only use `compute()` when necessary:

```suggestion
                result = np.allclose(
                    self.core_data(),
                    other.core_data(),
                    equal_nan=True,
                )
                result = _lazy.as_concrete_data(result)
```

Reply:
That would change the resulting datatype from a `bool` to a `numpy.ndarray` though. Would the solution in https://github.com/SciTools/iris/pull/5713/commits/ec30de862338e6e6b0dbb2e6032731e36b55cdbd also work for you?